app/models/TraderProfile.scala

```diff
@@ -80,4 +81,27 @@ object TraderProfile {
         }
       case Left(errors) => Left(errors)
     }
+
+  def validateHasNirms(
+    answers: UserAnswers
+  ): EitherNec[ValidationError, Boolean] =
+    answers.getPageValue(HasNirmsUpdatePage) match {
```

Comment:
does this mean you will get an unexpected page error if hasNirms is true? Surely that is an acceptable outcome as long as nirms number is present and remove page is false or empty

Reply:
Yes, But when answer yes it is part of the NirmsNumber update journey.

Reply:
May be we should change the method name?